### PR TITLE
V3.0.0 Chain upgrade

### PR DESCRIPTION
**Pull request by arhamchordia**
## 1. Overview

This PR upgrades the Quasar chain from Cosmos SDK version 0.47 to version 0.50.x. This includes updating dependencies, modifying import paths, and adjusting the codebase to align with the changes and improvements introduced in the newer SDK version.

## 2. Implementation details
Ref for the cosmos-sdk upgrade - https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/UPGRADING.md](https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/UPGRADING.md

New modules added :
- PFM v8
- Ibc-hooks v8
- Rate-limiting v8
- Feemarket v1

Updates : 
- Wasmd v0.45 to v0.51
- Wasmvm from v1 to v2
- Cosmos-SDK from v47.12 to v50.9
- Ibc-go v7 to ibc-go v8
- Contains updates of Async ICQ, wasm light clients and capability module.

Upgrade removes old unused modules : 
- `qVesting`
- `qOracle`
- `qTransfer`

## 3. How to test/use

There is a demos/upgrade-handler/v3.0.0 directory. 
Here, run the upgrade_test.sh to run an upgrade and then then tests like params checking, wasm, bank, staking, IBC txns. 

## 4. Checklist

- [x] Does the Readme need to be updated? YES

## 5. Limitations (optional)

The upgrade may introduce breaking changes requiring dependent services or module modifications.
Extensive testing is needed to ensure compatibility with existing data and user workflows.

## 6. Future Work (optional)

Add Slinky integration which will be followed on chain after v50 upgrade.

### CODE REVIEW

**Review by ajansari95 — DISMISSED** [review]
lgtm

File: app/upgrades/v2/upgrades.go [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
-	qoraclemoduletypes "github.com/quasarlabs/quasarnode/x/qoracle/types"
-	qtransfertypes "github.com/quasarlabs/quasarnode/x/qtransfer/types"
-	tftypes "github.com/quasarlabs/quasarnode/x/tokenfactory/types"
-
-	"github.com/quasarlabs/quasarnode/app/keepers"
-	"github.com/quasarlabs/quasarnode/app/upgrades"
+	"github.com/quasar-finance/quasar/app/keepers"
 )
 
+// qosmotypes "github.com/quasar-finance/quasar/x/qoracle/osmosis/types"
```
Comment (akure) [inline app/upgrades/v2/upgrades.go:11]: We can remove these comments now.

File: app/upgrades/v2/upgrades.go [hunk 2]
```diff
@@ -1,34 +1,31 @@
...
+	configurator module.Configurator,
+	keepers *keepers.AppKeepers,
+) upgradetypes.UpgradeHandler {
+	return func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+
+		return mm.RunMigrations(ctx, configurator, fromVM)
+	}
+}
+
+/*
```
Comment (akure) [inline app/upgrades/v2/upgrades.go:28]: Also  these commented code can be removed now

Comment (ajansari95) [reply app/upgrades/v2/upgrades.go]: [dc70348](https://github.com/quasar-finance/quasar/pull/791/commits/dc70348ef20411e7db24ca5f02692af30c04f8b3)

File: app/upgrades/v2/upgrades.go (hunk 1 above)
Comment (ajansari95) [reply app/upgrades/v2/upgrades.go]: [dc70348](https://github.com/quasar-finance/quasar/pull/791/commits/dc70348ef20411e7db24ca5f02692af30c04f8b3)

**Review by ajansari95 — APPROVED** [review]

**Review by akure — APPROVED** [review]
lgtm

**Pull request merged**